pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -683,6 +689,42 @@ func (mi *StorageInitializerInjector) createModelContainer(image string, modelPa
 	return modelContainer
 }
 
+func (mi *StorageInitializerInjector) createModelInitContainer(image string) *v1.Container {
+	cpu := mi.config.CpuModelcar
+	if cpu == "" {
+		cpu = CpuModelcarDefault
+	}
+	memory := mi.config.MemoryModelcar
+	if memory == "" {
+		memory = MemoryModelcarDefault
+	}
+
+	modelContainer := &v1.Container{
+		Name:  ModelcarInitContainerName,
+		Image: image,
+		Args: []string{
+			"sh",
+			"-c",
+			// Check that the expected models directory exists
+			"[ -d /models ] && [ \"$$(ls -A /models)\" ] && echo 'Found /models' || (echo 'Path /models does not exist or is empty' && exit 1)",
```

Comment:
Nice! This also helps validate a model car image early on. I'm not fully sure what happens if the init-container exits with 1, and if the the full Pod is restarted or just marked with an error. I would avoid that as it does not make sense to restart with the same image as the error won't go away.

Reply:
Can you also add some more context information to the error message? For example, the image name should be added here, too. That makes it easier to troubleshoot.

Reply:
You could also add the purpose of the container to the success message, so that people understand why this dummy container is started overall.

Something like:
```suggestion
			"echo 'Pre-fetching modelcar " + image + ": ' && [ -d /models ] && [ \"$$(ls -A /models)\" ] && echo 'OK ... Prefetched and valid (/models exists)' || (echo 'NOK ... Prefetched but modelcar is invalid (/models does not exist or is empty)' && exit 1)",
```

Reply:
I simply used your suggestion :-)

Reply:
> I'm not fully sure what happens if the init-container exits with 1, and if the the full Pod is restarted or just marked with an error.

In my tests, the cluster restarts the init container, and the pod persists. It will keep restarting the init contaienr, and will mark the pod with `Init:Error` status. After some additional restarts, the status is downgraded to a CrashLoopBackOff.

Reply:
Do you think it would be possible that if the init-container fails because e.g. the format is wrong, that we could bring the whole Pod in an error status ? That would be a nice to have, though. I think we can live with the current solution.